tests/link/pytorch/test_basic.py

```diff
@@ -0,0 +1,175 @@
+    py_res = pytensor_py_fn(*test_inputs)
+
+    if len(fgraph.outputs) > 1:
+        for j, p in zip(pytorch_res, py_res):
+            assert_fn(j, p)
+    else:
+        assert_fn(pytorch_res, py_res)
+
+    return pytensor_torch_fn, pytorch_res
+
+
+def test_pytorch_FunctionGraph_once():
+    """Make sure that an output is only computed once when it's referenced multiple times."""
+    from pytensor.link.pytorch.dispatch import pytorch_funcify
+
+    x = vector("x")
+    y = vector("y")
+
+    class TestOp(Op):
+        def __init__(self):
+            self.called = 0
+
+        def make_node(self, *args):
+            return Apply(self, list(args), [x.type() for x in args])
+
+        def perform(self, inputs, outputs):
+            for i, inp in enumerate(inputs):
+                outputs[i][0] = inp[0]
+
+    @pytorch_funcify.register(TestOp)
+    def pytorch_funcify_TestOp(op, **kwargs):
+        def func(*args, op=op):
+            op.called += 1
+            return list(args)
+
+        return func
+
+    op1 = TestOp()
+    op2 = TestOp()
+
+    q, r = op1(x, y)
+    outs = op2(q + r, q + r)
+
+    out_fg = FunctionGraph([x, y], outs, clone=False)
+    assert len(out_fg.outputs) == 2
+
+    out_torch = pytorch_funcify(out_fg)
+
+    x_val = torch.tensor([1, 2]).to(getattr(torch, config.floatX))
+    y_val = torch.tensor([2, 3]).to(getattr(torch, config.floatX))
+
+    res = out_torch(x_val, y_val)
+    assert len(res) == 2
+    assert op1.called == 1
+    assert op2.called == 1
+
+    res = out_torch(x_val, y_val)
+    assert len(res) == 2
+    assert op1.called == 2
+    assert op2.called == 2
+
+
+def test_shared():
+    a = shared(np.array([1, 2, 3], dtype=config.floatX))
+    pytensor_torch_fn = function([], a, mode="PYTORCH")
+    pytorch_res = pytensor_torch_fn()
+
+    assert isinstance(pytorch_res, torch.Tensor)
+    np.testing.assert_allclose(pytorch_res, a.get_value())
+
+    pytensor_torch_fn = function([], a * 2, mode="PYTORCH")
+    pytorch_res = pytensor_torch_fn()
+
+    assert isinstance(pytorch_res, torch.Tensor)
+    np.testing.assert_allclose(pytorch_res, a.get_value() * 2)
+
+    new_a_value = np.array([3, 4, 5], dtype=config.floatX)
+    a.set_value(new_a_value)
+
+    pytorch_res = pytensor_torch_fn()
+    assert isinstance(pytorch_res, torch.Tensor)
+    np.testing.assert_allclose(pytorch_res, new_a_value * 2)
+
+
+def test_shared_updates():
+    a = shared(0)
+
+    pytensor_torch_fn = function([], a, updates={a: a + 1}, mode="PYTORCH")
+    res1, res2 = pytensor_torch_fn(), pytensor_torch_fn()
+    assert res1 == 0
+    assert res2 == 1
+    assert a.get_value() == 2
+
+    a.set_value(5)
+    res1, res2 = pytensor_torch_fn(), pytensor_torch_fn()
+    assert res1 == 5
+    assert res2 == 6
+    assert a.get_value() == 7
+
+
+def test_pytorch_checkandraise():
+    p = scalar()
+    p.tag.test_value = 0
+
+    res = assert_op(p, p < 1.0)
+
+    function((p,), res, mode=pytorch_mode)
```

Comment:
This is not testing the behavior of check_and_raise works

Reply:
The jax implementation is also similar to this
https://github.com/pymc-devs/pytensor/blob/dbe0e09ae2a5f5d0119e0378259f843a95d36abc/tests/link/jax/test_basic.py#L199-L207